### ISSUE TITLE
Show new fields in course view

### ISSUE THREAD

**Issue by HoracioAlvarado** [PoetOS/moodle-local_metadata#17]
Hello,

We are trying to use this moodle plugin to add new fields to all courses. 
We were able to add the fields and set the values in each course, but we can't find how or where are the fields show.

Moodle 3.6

Any help would be appreciated.

**Comment by seb-vial** [comment]
Hello,

If your database is correctly filled with metadata for your courses, it's then up to you to implement the display feature in the theme you use.

**Comment by HoracioAlvarado** [comment]
@Birssan thanks for your answer.

We were looking for a plugin that allowed us to add new fields to courses without the need to modify moodle core code (we have done this in the past and updating moodle after is a nightmare)

As I understand, with this plugin we can add new fields and we only need to modify the theme to show the fields. 

A related question (please let me know if I need to open a new issue for this), we are going to need to get the field via Rest API, do you guys have any advise on this?

Thanks

**Comment by seb-vial** [comment]
That's right, as far as I know (and I could definitely be wrong) , there is no way for local plugins to "inject" view code in theme templates. You have to edit your theme templates and renderers to get metadata and display them. I would suggest (if not already done) to inherit from the theme you're using.

Regarding your need for REST API, I don't think this is implemented for this plugin, but if you really need that feature, you're welcome to check the developer documentation about [web services](https://docs.moodle.org/dev/Web_services) and more specifically [how to add web services to your plugin](https://docs.moodle.org/dev/Adding_a_web_service_to_a_plugin). If you decide to contribute to the plugin, don't hesitate to make a PR @mchurchward will look at it when he can.
